components/email-form.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { Link } from 'react-router';
+import validateEmailForm from './validate-email-form';
+
+export default class EmailForm extends React.Component {
+  constructor(props) {
+    super(props);
+    const { success, errors, value } = props;
+    this.state = {
+      success,
+      errors,
+      value: value || ''
+    };
+    this.formSubmit = this.formSubmit.bind(this);
+    this.inputChange = this.inputChange.bind(this);
+    this.resetForm = this.resetForm.bind(this);
+  }
+
+  formSubmittedSuccessfully() {
+    return this.state.success === true;
+  }
+
+  inputChange(e) {
+    this.setState({
+      value: e.target.value
+    });
+  }
+
+  resetForm(e) {
+    e.preventDefault();
+    this.setState({
+      success: undefined,
+      errors: [],
+      value: ''
+    });
+  }
+
+  formSubmit(e) {
+    e.preventDefault();
+    const emailValue = this.state.value;
+    // note how this is the same validator we ran on the server!
+    const result = validateEmailForm(emailValue);
+    this.setState({
+      errors: result.errors,
+      success: result.valid
+    });
+  }
+
+  renderErrorsList() {
+    return this.state.errors.map((err) => {
+      return (
+        <li className="form-error" key={err}>
+          <p>{err}</p>
+        </li>
+      );
+    });
+  }
+
+  renderForm() {
+    return (
+      <form method="post" action="/submit-email-form" onSubmit={this.formSubmit}>
```

Comment:
In a non-JS world the form will submit like a regular form would to `/submit-email-form`, but in a JS world React will capture it.